allofplos/article_class.py

```diff
@@ -199,26 +202,37 @@ def get_element_xpath(self, tag_path_elements=None, remote=False):
         :param caps_fixed: whether to render 'PLOS' in the journal name correctly, or as-is ('PLoS')
         :return: PLOS journal name at specified xpath location
         """
-        try:
-            journal = self.get_element_xpath(tag_path_elements=["/",
-                                                                "article",
-                                                                "front",
-                                                                "journal-meta",
-                                                                "journal-title-group",
-                                                                "journal-title"])
-            journal = journal[0].text
-        except IndexError:
-            journal_meta = self.get_element_xpath(tag_path_elements=["/",
-                                                                     "article",
-                                                                     "front",
-                                                                     "journal-meta"])
-            for journal_child in journal_meta[0]:
-                if journal_child.attrib['journal-id-type'] == 'nlm-ta':
-                    journal = journal_child.text
-                    break
+        journal = ''
+        journal_path_1 = self.get_element_xpath(tag_path_elements=["/",
+                                                                   "article",
+                                                                   "front",
+                                                                   "journal-meta",
+                                                                   "journal-title-group",
+                                                                   "journal-title"])
+        if len(journal_path_1):
+            journal = journal_path_1[0].text
+        else:
+            journal_path_2 = self.get_element_xpath(tag_path_elements=["/",
+                                                                       "article",
+                                                                       "front",
+                                                                       "journal-meta",
+                                                                       "journal-title"])
+            if len(journal_path_2):
+                journal = journal_path_2[0].text
+
+            else:
+                journal_meta = self.get_element_xpath(tag_path_elements=["/",
+                                                                         "article",
+                                                                         "front",
+                                                                         "journal-meta"])
+                for journal_child in journal_meta[0]:
+                    if journal_child.attrib['journal-id-type'] == 'nlm-ta':
+                        journal = journal_child.text
+                        break
```

Comment:
can you guarantee that there is only one instance of this journal-id-type? Right now it'll break as soon as it hits the first success